workflow/controller/pod_cleanup.go

```diff
@@ -1,21 +1,104 @@
 package controller
 
 import (
+	"context"
+
 	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
 
 	"github.com/argoproj/argo-workflows/v3/workflow/common"
+	"github.com/sirupsen/logrus"
 
 	wfv1 "github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1"
 	"github.com/argoproj/argo-workflows/v3/workflow/controller/indexes"
 )
 
-func (woc *wfOperationCtx) queuePodsForCleanup() {
+func (woc *wfOperationCtx) isImmediateDeletionStrategy() bool {
+	return woc.orig.Spec.PodGC != nil && woc.orig.Spec.PodGC.Strategy == wfv1.PodGCOnWorkflowCompletion
```

Comment:
This should change to something else, what should it be?

Reply:
I think we should add `PodGCImmediate` as a GC strategy.

Reply:
`OnPodCompletion` is already a strategy that I'd consider to be immediate

Reply:
Current options are here: https://argoproj.github.io/argo-workflows/fields/#podgc